site2/website/release-notes.md

```diff
@@ -1,6 +1,440 @@
+- Set and return topic names on message API. [11743](https://github.com/apache/pulsar/pull/11743)
+- PIP-91: Separate lookup timeout from operation timeout. [11627](https://github.com/apache/pulsar/pull/11627)
+- Switch from pretty print to compact print for configs. [11609](https://github.com/apache/pulsar/pull/11609)
+- Java Client: remove usage of reflection while using Pulsar Implementation classes. [11636](https://github.com/apache/pulsar/pull/11636)
+- Add a optional params scope for pulsar oauth2 client. [11931](https://github.com/apache/pulsar/pull/11931)
+
+### C++ Client
+- Support configuring optional scope field for OAuth2 authentication. [12305](https://github.com/apache/pulsar/pull/12305)
+- Support setting priority for the consumer. [12076](https://github.com/apache/pulsar/pull/12076)
+- Fixed attempting to connect to multiple IP addresses. [11958](https://github.com/apache/pulsar/pull/11958)
+- Handle error when shutting down client after forks. [11954](https://github.com/apache/pulsar/pull/11954)
+- Move all C symbols into C++ pulsar namespace. [11919](https://github.com/apache/pulsar/pull/11919)
+- Expose getLastMessageId in the Reader API. [11723](https://github.com/apache/pulsar/pull/11723)
+- Add padding characters to base64 encoded protobuf native schema. [11492](https://github.com/apache/pulsar/pull/11492)
+
+### Python Client
+- Do not sort schema fields by default. [12232](https://github.com/apache/pulsar/pull/12232)
+- Support setting namespace for python schema. [12175](https://github.com/apache/pulsar/pull/12175)
+- Expose Client.shutdown() method. [11955](https://github.com/apache/pulsar/pull/11955)
+
+### Proxy
+- set default httpProxyTimeout to 5 minutes. [12299](https://github.com/apache/pulsar/pull/12299)
+- Fixed NPE in ProxyConnection with no auth data. [12111](https://github.com/apache/pulsar/pull/12111)
+- [Issue-11966] set default http proxy request timeout. [11971](https://github.com/apache/pulsar/pull/11971)
+- Fixed Proxy leaking oubound connections. [11848](https://github.com/apache/pulsar/pull/11848)
+
+### Metrics
+- Add support for splitting topic and partition label in Prometheus. [12225](https://github.com/apache/pulsar/pull/12225)
+
+### Library updates
+- [security] Upgrade netty to 4.1.68.Final. [12218](https://github.com/apache/pulsar/pull/12218)
+- Added JLine 2.x for ZK CLI tool. [12102](https://github.com/apache/pulsar/pull/12102)
+- upgrade aircompressor to 0.20. [11790](https://github.com/apache/pulsar/pull/11790)
+- [Security] Upgrade Jetty to 9.4.43.v20210629. [11660](https://github.com/apache/pulsar/pull/11660)
+- [Security] Upgrade commons-compress to 1.21. [11345](https://github.com/apache/pulsar/pull/11345)
+- [Security] Bump Netty version to 4.1.66.Final. [11344](https://github.com/apache/pulsar/pull/11344)
+
+### Tiered Storage
+- Fix the potential race condition in the BlobStore readhandler. [12123](https://github.com/apache/pulsar/pull/12123)
+
+### Misc
```

Comment:
I remember that you've made a comment about `Misc`. If this part is not categorized into several small sections, the readability is not good since users do not have enough patience to wade through a large chunk of information.

Reply:
I believe that this part is pretty useless for users and we could remove it at all.

Having such detailed list of PRs is not useful for anyone.
Advanced users use "git log" or they can always use the GH interface to run thru the list of PRs.

I am sorry but I don't want to spend hours in classifying 2-3 hundreds of PRs

Reply:
Retaining some important PRs in `Misc` might be a good approach. 
Besides, it is better to invite other people to review this PR and give their opinions.

Reply:
I agree that `Misc` part is useless for users and we should remove it. See https://pulsar.apache.org/en/release-notes/#apache-pulsar-release-notes for example, the release notes of 2.8.1 and 2.8.0 don't contain this part.

But I also found many PRs in this part should be noted in other parts. For example,

```
- fix publish_time not set error when broker entry metadata enable without AppendBrokerTimestampMetadataInterceptor. [11014](https://github.com/apache/pulsar/pull/11014)
```

fixes a very common issue of Pulsar 2.8.0 when enabling KoP.

> I am sorry but I don't want to spend hours in classifying 2-3 hundreds of PRs

It might took some time but I think it's necessary. From my perspective, the release notes of this PR looks very terrible, especially compared with the existing release notes for 2.8.0 and 2.8.1.

Some semantics errors or inconsistency are acceptable. Take release notes of 2.8.1 as example,

```
Add metrics for writing or reading size of cursor #11500
Added missing configuration entries #11095
```

It's totally acceptable IMO.

But I think the PR description (in this PR) like

```
- fix lost message issues 12221. [12223](https://github.com/apache/pulsar/pull/12223)
```

should be fixed. No one knows what's the issue of 12221 only from the website.

The readers of release notes are usually those who want to check if this release has fixed some specific bugs or supported some features. The classification is very important though it could take much time.

Reply:
The `fix lost message issues 12221` title is a bad work done by who merged this PR. But we cannot avoid this problem since we have many committers now. I think release manager is responsible to do the cleanup for it.

Maybe we can add some enhancement for the process of merging PRs in future, but currently we should do it manually.